### PR TITLE
Dashboard testing

### PR DESCRIPTION
**Pull request by paulauus**
Added tests for the three function files for dashboard. All are passing, let me know if any weird issues. Everything is linted as well.
Also added a bash script for dockerising - basically same as the other ones you've done.

### CODE REVIEW

**Comment by paulauus** [comment]
Removing those testing issues was actually insane.

File: dashboard/README.md [hunk 1]
```diff
@@ -93,23 +93,21 @@ Build, tag, push, and deploy your Dockerised dashboard on AWS:
...
-aws ecr get-login-password --region <region> | docker login --username AWS --password-stdin <aws-account-id>.dkr.ecr.<region>.amazonaws.com
+AWS_ACCOUNT_ID=<Your AWS account ID>
+ECR_REPO_NAME=<Your ECR repository name>
 ```
-5. Push the image to ECR
+
+3. Run `push_to_ecr.sh`:
 ```
-docker push <aws-account-id>.dkr.ecr.<region>.amazonaws.com/<repository-name>:latest
+bash push-to-ecr.sh
```
Comment (Leon-JS02) [inline dashboard/README.md:107]: This is a lot simpler now!

File: dashboard/README.md [hunk 2]
```diff
@@ -93,23 +93,21 @@ Build, tag, push, and deploy your Dockerised dashboard on AWS:
...
 ```
-6. (Optional) Run the Docker Container Locally
+
```
Comment (Leon-JS02) [inline dashboard/README.md:109]: Nice to include this step

File: dashboard/pages/utils/test_subscribe_functions.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+from unittest.mock import patch
+import pandas as pd
+from subscribe_functions import (
+    get_subscriber_counts,
+    subscribe_user_to_topics,
+    unsubscribe_user_from_all_topics,
+)
+
+@pytest.fixture
+def mock_boto3_client():
```
Comment (Leon-JS02) [inline dashboard/pages/utils/test_subscribe_functions.py:14]: Nice workaround

**Review by Leon-JS02 — APPROVED** [review]
Good range of test cases covered, and good changes.

File: dashboard/push_to_ecr.sh [hunk 4]
```diff
@@ -0,0 +1,7 @@
+source .env
```
Comment (MichaelHassan) [inline dashboard/push_to_ecr.sh:1]: good bash script, makes the whole process easier.

**Review by MichaelHassan — APPROVED** [review]
Good testing and nice work patching with boto3

**Pull request merged**